ddprof-lib/src/main/cpp/javaApi.cpp

```diff
@@ -123,22 +123,40 @@ Java_com_datadoghq_profiler_JavaProfiler_getSamples(JNIEnv *env,
   return (jlong)Profiler::instance()->total_samples();
 }
 
+// some duplication between add and remove, though we want to avoid having an extra branch in the hot path
 extern "C" DLLEXPORT void JNICALL
-Java_com_datadoghq_profiler_JavaProfiler_filterThread0(JNIEnv *env,
-                                                       jobject unused,
-                                                       jboolean enable) {
-  int tid = ProfiledThread::currentTid();
-  if (tid < 0) {
+Java_com_datadoghq_profiler_JavaProfiler_filterThread_1add(JNIEnv *env,
```

Comment:
Here is actually a typo `1_add` this will not be linked with the Java native method.

Reply:
This is probably due to the mangling (I had to iterate a few times before I figured out how to link)